docker/stan-wasm-server/logic/authorization.py

```diff
@@ -0,0 +1,20 @@
+import os
+
+def _passcode_is_valid(passcode: str):
+    SWS_PASSCODE = os.environ.get("SWS_PASSCODE", "")
+    if not SWS_PASSCODE:
+        raise ValueError("SWS_PASSCODE environment variable not set")
```

Comment:
True, but it's cheap, and avoids awkward scoping workarounds.

Would you feel better about:
```python
def _passcode_is_valid(passcode: str):
    attrs = vars(_passcode_is_valid)
    if attrs.setdefault("SWS_PASSCODE", None) is None:
        SWS_PASSCODE = os.environ.get("SWS_PASSCODE", "")
        if not SWS_PASSCODE:
            raise ValueError("SWS_PASSCODE environment variable not set")

    return passcode == attrs["SWS_PASSCODE"]
```

or something like that?

Reply:
self modifying code also seems... not ideal.

Is there a reason to not just load it in the global scope and fail at startup, rather than the first time it gets queried?

Reply:
Well, doing so would make it less modular. It also restricts the possible entry points for the logic?

You're saying set it in `main.py` in global scope and it should still be visible within the module? (This isn't something I've done a whole lot in Python, so I wasn't sure if that would actually work.) Would it require any hoop-jumping to test?

Reply:
I'm saying move the lines

```
SWS_PASSCODE = os.environ.get("SWS_PASSCODE", "")
if not SWS_PASSCODE:
    raise ValueError("SWS_PASSCODE environment variable not set")
```
to just above the function, still in `authorization.py`.

This would make importing from `authorization` fail if it wasn't set, which would be at startup for our purposes. 

Testing would require some hoop-jumping, but not much more compared to the mocking of environment variables